index.php

```diff
@@ -0,0 +1,170 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Web interface for generating plugins.
+ *
+ * @package    tool_pluginkenobi
+ * @copyright  2016 Alexandru Elisei
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+use Monolog\Logger;
+use Monolog\Handler\BrowserConsoleHandler;
+
+require(__DIR__.'/../../../config.php');
+require_once($CFG->libdir.'/adminlib.php');
+require_once($CFG->libdir.'/moodlelib.php');
+require_once(__DIR__.'/vendor/autoload.php');
+
+admin_externalpage_setup('tool_pluginskel');
+
+$url = new moodle_url('/admin/tool/pluginskel/index.php');
+$PAGE->set_url($url);
+$PAGE->set_title(get_string('generateskel', 'tool_pluginskel'));
+$PAGE->set_heading(get_string('generateskel', 'tool_pluginskel'));
+
+$step = optional_param('step', '0', PARAM_INT);
+$component = optional_param('component1', '', PARAM_TEXT);
+
+$returnurl = new moodle_url('/admin/tool/pluginskel/index.php');
+
+if ($step == 0) {
+
+    $mform0 = new tool_pluginskel_step0_form();
+    $formdata = $mform0->get_data();
+    $recipefile = $mform0->get_file_content('recipefile');
+
+    if (!empty($formdata)) {
+        $data = array();
+
+        if (empty($recipefile)) {
+            if (empty($formdata->componentname)) {
+                throw new moodle_exception('emptypluginname', 'tool_pluginskel', $returnurl);
+            }
+            $recipe['component'] = $formdata->componenttype.'_'.$formdata->componentname;
+        } else {
+            $recipe = tool_pluginskel\local\util\yaml::decode_string($recipefile);
+            $templatevars = tool_pluginskel\local\util\manager::get_component_variables($component);
+
+            foreach ($templatevars as $var) {
+                if ($var['hint'] == 'multiple') {
+                    $formvariable = $var['name'].'count';
+                    if (empty($recipe[$var['name']])) {
+                        $count = 1;
+                    } else {
+                        $count = count($recipe[$var['name']]);
+                    }
+                    $data[$formvariable] = $count;
```

Comment:
I need this for variables that have an array as value (like dependencies). I don't know how many elements are in the array, so I need to save their number when creating the form.